chore(website): add responsive layout and vertical examples (#1756)
twilio-labs/paste: packages/paste-website/src/pages/layout/grid/index.mdx

### Responsive columns

> Grid column `span`, `offset`, and `gutter` props accept accept responsive values, allowing them to be changed on theme breakpoints.

Reviewer: ```suggestion
Grid column `span`, `offset`, and `gutter` props accept responsive values, allowing them to be changed on theme breakpoints.
```

> For example, passing `span={[6, 8, 2] }` to a `Column`component sets the column `span` to be 6 columns wide on mobile, 8 columns wide on tablet, and 2 columns wide on desktop. The `offset`, `gutter`, and `vertical` props work in a similar fashion.

Reviewer: ```suggestion
For example, setting `span={[6, 8, 2] }` on a `Column` will instruct the column to `span` 6 columns of the grid at the first breakpoint, 8 columns at the second, and 2 columns at the third breakpoint.
```

Check out the examples below and resize the browser to see the responsive values in action.

> While you can achieve a one column responsive layout by changing each column `span` to 12, you lose out on the ability to have vertical spacing between each row. In this case, we recommend using the vertical prop as it will continue to apply the gutter spacing

Reviewer: ```suggestion
When creating responsive layouts with the Grid, it is important to remember to control the `vertical` prop accordingly. At the breakpoints you wish to stack the columns, set `vertical` to equal true. This will apply the column gutters vertically, rather than horizontally.
```

Author: @SiTaggart I replaced the text here, and moved a slightly reworded call out about setting span to 12 in a callout. LMK what you think, can easily remove.

Author: Here is is with the callout 👇 

<img width="816" alt="Screen Shot 2021-08-11 at 6 43 48 AM" src="https://user-images.githubusercontent.com/18120906/129040326-9f280b1a-07f8-4d4b-8ca9-3e8be68401cf.png">

Author: (btw I see that there is a mis-formatted word "span"), if you want me to include this I can fix it when I restore the commit)

Author: Okey cool, This is ready to go then for another review. The most current commit doesn't have the callout.

  </Dont>
</DoDont>

<DoDont>
  <Do title="Do" body="Use a vertical grid to create a a stack columns.">
    <Box padding="space60">
      <Grid gutter="space30" vertical>
        <Column>
          <Card>
            <Paragraph>Card content</Paragraph>
          </Card>
        </Column>
        <Column>
          <Card>
            <Paragraph>Card content</Paragraph>
          </Card>
        </Column>
      </Grid>
    </Box>
  </Do>
  <Dont title="Don't" body="Don’t use column span to stack columns.">
    <Box padding="space60">
      <Grid gutter="space30">
        <Column span={12}>
          <Card>
            <Paragraph>Card content</Paragraph>
          </Card>
        </Column>
        <Column span={12}>
          <Card>
            <Paragraph>Card content</Paragraph>
          </Card>
        </Column>
        <Column span={12}>
          <Card>
            <Paragraph>Card content</Paragraph>
          </Card>
        </Column>
      </Grid>
    </Box>
  </Dont>
</DoDont>

> 

Reviewer: Let's just remove this do don't. 

If you wanna  set column spans to 12 by all means do so. you might have a good reason.
